### PR TITLE
Rename R function from AppendSlopeFeatures to append_slope_features

### PR DESCRIPTION
**Pull request by jayqi**
I propose we rename the R function `AppendSlopeFeatures` to `append_slope_features` to match the interface of the Python package. 

Lower-snake-case is also a commonly used style in R, e.g., see [Tidyverse style guide](https://style.tidyverse.org/syntax.html), so I think most R users may find this change to even be an improvement. 

This PR makes progress towards #13

### CODE REVIEW

File: r-client/R/client.R [hunk 1]
```diff
@@ -16,9 +15,9 @@
 #'                          in degrees. If not supplied, this will be calculated
 #'                          from consecutive observations}
 #' }
-#' @param hostName A string with the host running groundhog. By default, this
```
Comment (Haoen-Cui) [inline r-client/R/client.R:19]: @jayqi: can you ignore white space changes? 
Also occurred multiple times elsewhere.

Comment (jayqi) [reply r-client/R/client.R:19]: @Haoen-Cui if you click on the Gear icon when viewing the diff, you can hide whitespace changes in the UI.

Typically, we consider not-having-trailing-whitespace to be a good best practice, and we encourage people to turn on automatic-stripping in RStudio and other text editors.

Comment (jameslamb) [reply r-client/R/client.R:19]: yeah it's totally fine to have your editor touch these other lines. I'm cool with it.

**Review by jameslamb — APPROVED** [review]
I totally support this. Thanks for contribution @jayqi !

**Pull request merged**